### PR TITLE
Add rector config

### PR DESCRIPTION
**Pull request by come-nc**
## Summary

And so it begins…

1. Added rector in a vendor-bin
2. Added a mostly-empty configuration
3. Applied to apps folder

Long term goal is to apply rectors from https://github.com/nextcloud-libraries/rector to core code. We’ll need to be extra careful with folders like lib/public because they use some deprecated stuff on purpose I think.
So not sure if the best way forward is to add more rules or more folders.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [ ] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: rector.php [hunk 1]
```diff
@@ -20,6 +20,12 @@
 		// __DIR__ . '/tests',
 		// __DIR__ . '/themes',
 	])
+	->withSkip([
+		__DIR__ . '/apps/*/3rdparty/*',
+		__DIR__ . '/apps/*/build/stubs/*',
+		__DIR__ . '/apps/*/composer/*',
+		__DIR__ . '/apps/*/config/*',
+	])
```
Comment (provokateurin) [inline rector.php:28]: Please put this into the first commit

**Review by provokateurin — CHANGES_REQUESTED** [review]
Can you add a script like https://github.com/nextcloud/groupfolders/blob/532f734b9d729a3c29a7371faf0bb85a24e0b311/composer.json#L26? This makes it a lot easier to use rector.

File: rector.php [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+use Rector\Config\RectorConfig;
+
+return RectorConfig::configure()
+	->withPaths([
+		__DIR__ . '/apps',
```
Comment (provokateurin) [inline rector.php:14]: Make sure to exclude any git ignored paths like cs-fixer

**Review by provokateurin — CHANGES_REQUESTED** [review]

**Comment by provokateurin** [comment]
Uhm is it really supposed to run on 50323 files? To me it looks like it includes way too much.

**Comment by come-nc** [comment]
> Uhm is it really supposed to run on 50323 files? To me it looks like it includes way too much.

1877 for me

**Comment by provokateurin** [comment]
Maybe because I have my cloned apps in apps/ as well, and then it also scans all the vendors of those and so on.
When ignoring everything that git ignores it should work though.

**Review by provokateurin — APPROVED** [review]
Works, you only need to make the CI happy.

**Review by nickvergessen — APPROVED** [review]

**Pull request merged**